metadata/metadata.go

```diff
@@ -1511,6 +1759,66 @@ func (serv *MetadataServer) GetModels(stream pb.Metadata_GetModelsServer) error
 	})
 }
 
+func (serv *MetadataServer) GetEquivalent(ctx context.Context, req *pb.ResourceVariant) (*pb.ResourceVariant, error) {
+	return serv.getEquivalent(req, true)
+}
+
+/*
+*
+This method is used to get the equivalent resource variant for a given resource variant. readyStatus is used to determine
+if we should only return the equivalent resource variant if it is ready.
+*/
+func (serv *MetadataServer) getEquivalent(req *pb.ResourceVariant, filterReadyStatus bool) (*pb.ResourceVariant, error) {
+	noEquivalentResponse := &pb.ResourceVariant{}
+
+	var currentResource ResourceVariant
+	var resourceType ResourceType
+
+	switch req.Resource.(type) {
+	case *pb.ResourceVariant_SourceVariant:
+		currentResource = &sourceVariantResource{req.GetSourceVariant()}
+		resourceType = SOURCE_VARIANT
+	case *pb.ResourceVariant_FeatureVariant:
+		currentResource = &featureVariantResource{req.GetFeatureVariant()}
+		resourceType = FEATURE_VARIANT
+	case *pb.ResourceVariant_LabelVariant:
+		currentResource = &labelVariantResource{req.GetLabelVariant()}
+		resourceType = LABEL_VARIANT
+	case *pb.ResourceVariant_TrainingSetVariant:
+		currentResource = &trainingSetVariantResource{req.GetTrainingSetVariant()}
+		resourceType = TRAINING_SET_VARIANT
+	default:
+		return nil, fmt.Errorf("unknown resource variant type: %v", req.Resource)
+	}
+
+	resources, err := serv.lookup.ListForType(resourceType)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, res := range resources {
+		resourceStatus := res.GetStatus()
```

Comment:
Can you put the contents of this loop into its own func for readability?

Reply:
gladly